test/services/MailService.spec.ts

```diff
@@ -0,0 +1,56 @@
+import MailService from '@/services/MailService';
+import mail from '@test/data/mail';
+import message from '@test/data/message';
+import { expect } from 'chai';
+import 'mocha';
+import nodeMailer from 'nodemailer';
+
+describe('MailService.send', () => {
+
+  // Instantiate the service
+  const mailService = new MailService();
+  let messageInfo: any;
+
+  before('should send an e-mail.', async () => {
+    // Create ethereal mail account
+    const account = await nodeMailer.createTestAccount();
+    const transporter = nodeMailer.createTransport({
+      host: 'smtp.ethereal.email',
+      port: 587,
+      secure: false,
+      auth: {
+        user: account.user,
+        pass: account.pass,
+      },
+    });
+
+    // Add the custom transporter
+    (mailService as any).transporters.push({
+      platformId: message.platform,
+      transporter,
+    });
+
+    // Send a mail
+    messageInfo = await mailService.send(message.platform, mail);
+  });
+
+  it('should send an e-mail, accepted by the receiver.', async () => {
+    expect(messageInfo.accepted)
+      .to.be.an('array')
+      .to.have.lengthOf(1)
+      .to.include(mail.to);
+  });
+
+  it('should send an e-mail, which is not rejected.', async () => {
+    expect(messageInfo.rejected)
+      .to.be.an('array')
+      .that.is.empty
+      .to.not.include(mail.to);
+  });
+
```

Comment:
Can an email get accepted by the receiver but still rejected? Are these two really separate test cases?

Reply:
Good comment, I merged the two test cases.